### DISCUSSION TITLE
What improvements we can do in clean-jsdoc-theme?

### DISCUSSION THREAD

**Discussion by ankitskvmdam** [ankitskvmdam/clean-jsdoc-theme#108] [General]
Feel free to and all the improvements which we can do to improve clean-jsdoc-theme

**Comment by dineshdevlpr** [comment]
Hi Ankit, extraordinary work.

I've been wondering that if the sub-elements present inside of the particular file can also be listed in the Navigation bar, then it will become the better one among all the Jsdoc templates.

Is there any progression for the displaying the sub-elements in Navigation bar going on ? Then please let us know here.

Thanks for clean-jsdoc-theme. You've already nailed it.

**Reply by ankitskvmdam** [reply]
Could you please explain, what you mean by sub element?

Also do you want them as Dropdown Menu?

**Reply by dineshdevlpr** [reply]
Sure, What i wanted to convey is

Here --> https://ankdev.me/clean-jsdoc-theme/v4/Apple.html

In Navigation bar>Classes>Apple

When documentation for Apple is being displayed, The elements that apple class contains such as color, energy, gameQuery, NUMBER_ARRAY etc are there right.

What I was asking is, Is there any option to display those sub-elements under Each topics present in Navigation Bar itself via expandable dropdown like this --> http://clenemt.github.io/docdash/documents_model.html or like the Toast Ui(https://github.com/nhn/tui.jsdoc-template).

**Reply by ankitskvmdam** [reply]
It was there in v3. Now we have table of content(TOC) for each pages. TOC is only visible on large screens.

**Comment by brewhousedigital** [comment]
I manage a marketing site of ~400 pages built in NextJS. We only use Components and Hooks for the app. Is there a way to add custom "sections" to the system? 

Perhaps adding `@kind Component` or `@kind Hook` will allow for custom sections?

**Reply by ankitskvmdam** [reply]
@brewhousedigital Do you have any idea on how to do this?

**Reply by brewhousedigital** [reply]
Not sure. I tried to see how the library was collecting each of the `SECTION_TYPE`, but that might be a thing from `jsdoc/util/templateHelper`. 

No worries, I was mostly just curious.
